pkg/controller/flinkapplication/flink_state_machine.go

```diff
@@ -754,15 +755,52 @@ func (s *FlinkStateMachine) handleSubmittingJob(ctx context.Context, app *v1beta
 	if job == nil {
 		return statusUnchanged, errors.Errorf("Could not find job %s", s.flinkController.GetLatestJobID(ctx, app))
 	}
-
-	// wait until all vertices have been scheduled and started
-	allVerticesStarted := true
-	for _, v := range job.Vertices {
-		allVerticesStarted = allVerticesStarted && (v.StartTime > 0)
+	allVerticesRunning := true
+	if app.Spec.BetaFeaturesEnabled {
+		// wait until all vertices have been scheduled and running
+		logger.Info(ctx, "Beta features flag is enabled.")
+		jobStartTimeSec := job.StartTime / 1000
```

Comment:
Can we throw the below logic in a separate function to make this more readable and testable?

Reply:
I should have clarified that I meant all the new logic of the running vertex state check, not just the start time calculation

Reply:
After removing the feature flag if condition, I think it's readable now.

Reply:
func is >100 lines. I think it's somewhat difficult to read. Nit, but for maintainability